### PR TITLE
test (non)equality for alt_string implementation

### PR DESCRIPTION
**Pull request by agrianius**
Add a bit more tests for alt_string implementation

### CODE REVIEW

File: test/src/unit-alt-string.cpp [hunk 1]
```diff
@@ -60,17 +66,36 @@ class alt_string
     }
 
     template <typename op_type>
-    bool operator==(op_type&& op) const
+    typename std::enable_if< // disable for alt_string
+                !std::is_same<  alt_string,
+                                typename std::remove_reference<op_type>::type
+                             >::value,
+                bool>::type
```
Comment (theodelrieu) [inline test/src/unit-alt-string.cpp]: I think you can remove all of this and simply take the parameter as `const op_type&`.

Comment (agrianius) [reply test/src/unit-alt-string.cpp]: Yeap, you are right, thanx for the notice. I will make another request.

Comment (theodelrieu) [reply test/src/unit-alt-string.cpp]: No need to open a different PR by the way, pushing another commit will serve the same purpose.

Comment (agrianius) [reply test/src/unit-alt-string.cpp]: Oh, I did not know, thanx.

**Pull request closed without merging**